app/src/main/java/org/pragyan/dalal18/ui/RegistrationActivity.kt

```diff
@@ -86,6 +86,7 @@ class RegistrationActivity : AppCompatActivity() {
                             .setFullName(nameEditText.text.toString())
                             .setPassword(passwordEditText.text.toString())
                             .setUserName(nameEditText.text.toString())
+                            .setReferralCode(referralCodeEditText.text.toString())
```

Comment:
Need to add **InvalidReferralCodeError** also !

refer [this](https://github.com/delta/dalal-street-proto/blob/master/actions/Register.proto) file

Reply:
There are only 6 fields in Register Request proto file you linked above.
```
    string email = 1;
    string password = 2;
    string user_name = 3;
    string full_name = 4;
    string country = 5;
    string referral_code = 6;
```

Where is InvalidReferralCodeError? Check the generated .java file from this proto, there isn't any such field there either.

Reply:
> There are only 6 fields in Register Request proto file you linked above.
> 
> ```
>     string email = 1;
>     string password = 2;
>     string user_name = 3;
>     string full_name = 4;
>     string country = 5;
>     string referral_code = 6;
> ```
> 
> Where is InvalidReferralCodeError? Check the generated .java file from this proto, there isn't any such field there either.

not here, below this code where we have several response messages !

Reply:
Yeah got it, you pointed it at the request builder so got confused.